packages/editor/src/browser/editor-linenumber-contribution.ts

```diff
@@ -0,0 +1,95 @@
+import { FrontendApplicationContribution, QuickInputService, ApplicationShell, ContextMenuRenderer } from '@theia/core/lib/browser';
+import { ContextKeyService } from '@theia/core/lib/browser/context-key-service';
+import { DisposableCollection, MenuPath } from '@theia/core';
+import { EditorWidget } from './editor-widget';
+
+export const EDITOR_LINENUMBER_CONTEXT_MENU: MenuPath = ['editor_linenumber_context_menu'];
+
+@injectable()
+export class EditorLineNumberContribution implements FrontendApplicationContribution {
+
+    @inject(ApplicationShell)
+    protected readonly shell: ApplicationShell;
+
+    @inject(ContextKeyService)
+    protected readonly contextKeyService: ContextKeyService;
+
+    @inject(ContextMenuRenderer)
+    protected readonly contextMenuRenderer: ContextMenuRenderer;
+
+    @inject(EditorManager)
+    protected readonly editorManager: EditorManager;
+
+    @inject(QuickInputService) @optional()
+    protected readonly quickInputService: QuickInputService;
+
+    onStart(): void {
+        this.editorManager.onCreated(editor => this.addLineNumberContextMenu(editor));
+    }
+
+    protected readonly toDisposeOnClosedEditor = new DisposableCollection();
+
+    protected addLineNumberContextMenu(editorWidget: EditorWidget): void {
+        const editor = editorWidget.editor;
+        if (editor) {
+            this.toDisposeOnClosedEditor.push(
+                editor.onMouseDown(event => this.handleContextMenu(editor, event))
+            );
+            editorWidget.onDispose(() => this.toDisposeOnClosedEditor.dispose());
+        }
+    }
+
+    protected handleContextMenu(editor: TextEditor, event: EditorMouseEvent): void {
+        if (event.target && event.target.type === MouseTargetType.GUTTER_LINE_NUMBERS) {
+            if (event.event.button === 2) {
+                editor.focus();
+                const lineNumber = lineNumberFromPosition(event.target.position);
+                const contextKeyService = this.contextKeyService.createOverlay([['editorLineNumber', lineNumber]]);
+                const uri = editor.getResourceUri()!;
+                const args = [{
+                    lineNumber: lineNumber,
+                    uri: uri['codeUri']
+                }];
+
+                setTimeout(() => {
+                    this.contextMenuRenderer.render({
+                        menuPath: EDITOR_LINENUMBER_CONTEXT_MENU,
+                        anchor: event.event,
+                        args,
+                        contextKeyService,
+                        onHide: () => contextKeyService.dispose()
+                    });
+                });
+            }
+        }
+    }
+
+}
+
+function lineNumberFromPosition(position: Position | undefined): number | undefined {
+    // position.line is 0-based line position, where the expected editor line number is 1-based.
+    if (position) {
+        return position.line + 1;
+    }
+    return undefined;
+}
```

Comment:
That's why Theia has m2p and p2m converters.

Reply:
@kittaakos, thanks a lot for your review! 
As far as I can see, the ProtocolToMonacoConvrter is located on the `monaco` package, where the contribution is currently in the `editor` one. So I cannot rely on it. Is there another converter I am not aware of?

Reply:
> As far as I can see, the ProtocolToMonacoConvrter is located on the `monaco` package, where the contribution is currently in the `editor` one. So I cannot rely on it.

I see. Thanks for the explanation.

> Is there another converter I am not aware of?

I am unaware of another one, so let's not complicate things before the release. Please go ahead.

(On the other hand, the implementation could be in `monaco`. `event.target.type === MouseTargetType.GUTTER_LINE_NUMBERS` is monaco specific anyway.)

Reply:
Interesting, I cannot resolve my threads on GitHub anymore. Please mark it as resolved and proceed with the merge. Thank you for the updates, @rschnekenbu

Reply:
Thanks a lot for your review. I rebased my contribution on latest master and solved the conflicts.